forge-game/src/main/java/forge/game/ability/effects/AmassEffect.java

```diff
@@ -52,13 +57,20 @@ public void resolve(SpellAbility sa) {
 
         final int amount = AbilityUtils.calculateAmount(card, sa.getParamOrDefault("Num", "1"), sa);
         final boolean remember = sa.hasParam("RememberAmass");
+        final String type = sa.getParam("Type");
 
         // create army token if needed
         if (!Iterables.any(activator.getCardsIn(ZoneType.Battlefield), CardPredicates.isType("Army"))) {
             CardZoneTable triggerList = new CardZoneTable();
             MutableBoolean combatChanged = new MutableBoolean(false);
 
-            makeTokenTable(makeTokenTableInternal(activator, "b_0_0_zombie_army", 1, sa), false, triggerList, combatChanged, sa);
+            final Card result = TokenInfo.getProtoType("b_0_0_army", sa, activator, false);
```

Comment:
Incubate should probably also use this now right?
(Not call `protoTypeApplyTextChange` so the Phyrexian type isn't able to be modded by the creating effect)

Reply:
Incubate should already do that, but I will check that out later

Reply:
Yes Incubate does it, the token script is already protected there by card text changes